<?xml version="1.0" encoding="UTF-8"?>

<launch>

    <!-- these are the arguments you can pass this launch file, for example paused:=true -->

    <arg name="paused" default="false"/>
    <arg name="use_sim_time" default="true"/>
    <arg name="gui" default="true"/>
    <arg name="headless" default="false"/>
    <arg name="debug" default="false"/>

  <!-- We resume the logic in empty_world.launch, changing only the name of the world to be launched -->
  <include file="$(find gazebo_ros)/launch/empty_world.launch">
	<arg name="world_name" value="$(find solution_jibo_description)/world/room.world"/>

    <arg name="debug" value="$(arg debug)" />
    <arg name="gui" value="$(arg gui)" />
    <arg name="paused" value="$(arg paused)"/>
    <arg name="use_sim_time" value="$(arg use_sim_time)"/>
    <arg name="headless" value="$(arg headless)"/>
    <!-- more default parameters can be changed here -->
  </include>

    <arg name="urdf_robot_file" default="$(find solution_jibo_description)/urdf/jibo2.urdf" />
    <param name="robot_description" command="cat $(arg urdf_robot_file)" />

    <arg name="x" default="1.0" />
    <arg name="y" default="0.0" />
    <arg name="z" default="0.0" />
    <arg name="roll" default="0.0"/>
    <arg name="pitch" default="0.0"/>
    <arg name="yaw" default="0.0"/>

    <group ns="jibo2">
        <!-- Spawn Part -->

        <arg name="robot_name" default="jibo2" />


        <rosparam file="$(find solution_jibo_description)/config/jibo2_controllers.yaml" command="load" ns="/jibo2"/>

        <param name="/jibo2/robot_description" command="cat $(arg urdf_robot_file)" />

        <!-- send fake joint values -->
        <node name="joint_state_publisher" pkg="joint_state_publisher" type="joint_state_publisher">
        <param name="use_gui" value="TRUE"/>
        </node>


        <node name="urdf_spawner_jibo2" pkg="gazebo_ros" type="spawn_model" respawn="false" output="screen"
        args="-urdf -x $(arg x) -y $(arg y) -z $(arg z) -R $(arg roll) -P $(arg pitch) -Y $(arg yaw) -model $(arg robot_name) -param /jibo2/robot_description"/>

        <node name="robot_state_publisher_jibo2" pkg="robot_state_publisher" type="robot_state_publisher"
        respawn="false" output="screen">
            <param name="publish_frequency" type="double" value="5.0" />
        </node>

        <node name="jibo2_controller_spawner" pkg="controller_manager" type="spawner" respawn="false"
        output="screen" args="--namespace=/jibo2
                              /jibo2/joint_state_controller
                              /jibo2/base_waist_joint_position_controller
                              /jibo2/waist_body_joint_position_controller
                              /jibo2/body_head_joint_position_controller
                              --shutdown-timeout 3">
        </node>

	<!-- Show in Rviz -->
	<!--node name="rviz" pkg="rviz" args="-d $(find solution_jibo_description)/rviz_config/urdf.rviz"/>-->
	<node name="rviz" pkg="rviz" type="rviz" args=""/>


    </group>

</launch>
